commit ab6ef34b906546d85d92dbc3e0fb4e01cce05f62
Author: Avi Kivity <avi@qumranet.com>
Date:   Tue Oct 16 16:23:22 2007 +0200

    KVM: Move apic timer interrupt backlog processing to common code
    
    Beside the obvious goodness of making code more common, this prevents
    a livelock with the next patch which moves interrupt injection out of the
    critical section.
    
    Signed-off-by: Avi Kivity <avi@qumranet.com>

diff --git a/drivers/kvm/kvm_main.c b/drivers/kvm/kvm_main.c
index 5fd2864b7811..4c96817929fd 100644
--- a/drivers/kvm/kvm_main.c
+++ b/drivers/kvm/kvm_main.c
@@ -2144,6 +2144,8 @@ static int __vcpu_run(struct kvm_vcpu *vcpu, struct kvm_run *kvm_run)
 	if (unlikely(r))
 		goto out;
 
+	kvm_inject_pending_timer_irqs(vcpu);
+
 	preempt_disable();
 
 	kvm_x86_ops->prepare_guest_switch(vcpu);